### ISSUE TITLE
[Touch unit] (multi) touch detection

### ISSUE THREAD

**Issue by pakLebah** [tranleduy2000/pascalnide#5]
There's no Touch unit available yet. So, I propose to provide **Touch** unit to handle touch and gesture events. For example:
- single tap detection
- double tap detection
- tap-n-hold (long-tap) detection
- multi-touch (up to 4 touches) detection
- edge swipe detection
- long swipe detection
- rotate swipe detection
- pinch detection (for zoom)
- … and other common touch and gesture events.

**Comment by tranleduy2000** [comment]
**Pascal** is not the language used for event programming, I would push these events into a queue and you could get it with a function like ``getLastGesture``.

**Comment by pakLebah** [comment]
Wrong! It's **your** Pascal N-IDE language that is unable to handle event programming. [Lazarus IDE](https://www.getlazarus.org) (GUI IDE for [Free Pascal](http://freepascal.org)) and [Delphi](https://www.embarcadero.com/products/delphi) had been doing it since '90-ish. **Your** Pascal language —as I've been saying from the beginning— lacks of so many **modern** Pascal language features that makes Pascal programming with your app is so limiting. It's **your** fault as the creator, not Pascal. 

Modern Pascal can do anything other languages can do. You can write cross platform applications using RemObjects' [Oxygene](http://elementscompiler.com/elements/oxygene) i.e. for MacOS/iOS Cocoa, Windows .Net, and Java Android. You can even write front-end web applications using [Smart Pascal](http://smartmobilestudio.com) (that compiles Pascal codes into optimized and high performance JavaScript code). While **your** Pascal is hardly able to handle object oriented programming correctly. **Your** Pascal N-IDE app is good, but the language is like Pascal from the '80-ish… old and limited.

**Comment by tranleduy2000** [comment]
I am developing class type, I will support event programming in the future.
